### PR TITLE
refactor: move report within status page

### PR DESCRIPTION
**Pull request by mxkaske**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation

### Before this PR

{Please add a screenshot here}

### After this PR

{Please add a screenshot here}

### CODE REVIEW

File: apps/web/src/app/app/[workspaceSlug]/(dashboard)/status-pages/[id]/reports/[reportId]/overview/page.tsx [hunk 1]
```diff
@@ -36,7 +35,7 @@ export default async function OverviewPage({
           action={
             <Button asChild>
               {/* TODO: check if correct */}
-              <Link href={`./${params.id}/update/edit`}>Create</Link>
+              <Link href={`./${report.id}/update/edit`}>Create</Link>
```
Comment (thibaultleouay) [inline apps/web/src/app/app/[workspaceSlug]/(dashboard)/status-pages/[id]/reports/[reportId]/overview/page.tsx]: Edit in link and create the button 🤔

Comment (mxkaske) [reply apps/web/src/app/app/[workspaceSlug]/(dashboard)/status-pages/[id]/reports/[reportId]/overview/page.tsx]: Good call. Removed the `action`, we already have the create button.

Screenshot 2024-07-01 at 10 36 48 PM

File: apps/server/src/v1/statusReportUpdates/post.ts [hunk 2]
```diff
@@ -81,31 +80,24 @@ export function registerPostStatusReportUpdate(
 
     // send email
 
-    if (workspacePlan.title !== "Hobby") {
-      const allPages = await db
+    if (workspacePlan.title !== "Hobby" && _statusReport.pageId) {
```
Comment (mxkaske) [inline apps/server/src/v1/statusReportUpdates/post.ts]: Lets use the dynamic getLimit function.

```suggestion
    if (getLimit(workspacePlan, "status-subscribers") && _statusReport.pageId) {
```

File: apps/server/src/v1/statusReports/post.ts [hunk 3]
```diff
@@ -118,62 +80,26 @@ export function registerPostStatusReport(api: typeof statusReportsApi) {
...
-            return {
-              monitorId: id,
-              statusReportId: _newStatusReport.id,
-            };
-          })
-        )
-        .returning();
-    }
-
     if (workspacePlan.title !== "Hobby") {
```
Comment (mxkaske) [inline apps/server/src/v1/statusReports/post.ts]: Here too. Let's clean up and use dynamic `getLimit`.

File: apps/server/src/v1/statusReports/update/post.ts [hunk 4]
```diff
@@ -77,29 +76,23 @@ export function registerStatusReportUpdateRoutes(api: typeof statusReportsApi) {
       .returning()
       .get();
 
-    if (workspacePlan.title !== "Hobby") {
-      const allPages = await db
+    if (workspacePlan.title !== "Hobby" && _statusReport.pageId) {
```
Comment (mxkaske) [inline apps/server/src/v1/statusReports/update/post.ts]: `getLimit`

File: apps/web/src/app/app/[workspaceSlug]/(dashboard)/status-pages/[id]/reports/[reportId]/edit/page.tsx [hunk 5]
```diff
@@ -4,30 +4,21 @@ import { api } from "@/trpc/server";
...
+    id: Number.parseInt(params.reportId),
   });
 
-  const monitors = await api.monitor.getMonitorsByWorkspace.query();
-
-  const pages = await api.page.getPagesByWorkspace.query();
-
   return (
     <StatusReportForm
-      monitors={monitors}
```
Comment (mxkaske) [inline apps/web/src/app/app/[workspaceSlug]/(dashboard)/status-pages/[id]/reports/[reportId]/edit/page.tsx:19]: Why removing the monitors?

**Review by mxkaske — COMMENTED** [review]
We need to re-add the monitor connections.

CleanShot 2024-07-10 at 20 03 29@2x

**Pull request merged**